### PR TITLE
feat: Adding vale linter

### LINKED ISSUE #178: Feature: Add Documenation Linting to CI/CD
### Type of feature

📦 Chore

### Current behavior

Currently there is nothing checking for stuff like an acronym being used before it's defined in documentation.

I came across this while reviewing #177

### Suggested solution

It made me think of the docs linting we used at Netlify. I think this could be a great addition to maintaining our docs. More about it here, https://www.netlify.com/blog/a-key-to-high-quality-documentation-docs-linting-in-ci-cd/

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by CBID2**
## Description
This PR adds the code for the Vale linter GitHub Action to the `compliance.yml` file. This addition will ease the process of maintaining docs in regards to checking for Markdown errors. 

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [x] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents
Closes #178 

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [X] 🙋 no, because I need help

### Update 10/24/22
Per @nickytonline's suggestion, I'm testing [the Vale Linter GitHub action](https://github.com/marketplace/actions/vale-linter) in [this test repo](https://github.com/CBID2/github-actions-test).   So far, I used this method to place the action's suggested code into the `compliance.yml` file.

```
name: "Compliance"

on:
  pull_request_target:
    types:
      - opened
      - edited
      - synchronize

permissions:
  pull-requests: write

jobs:
  compliance:
    uses: open-sauced/hot/.github/workflows/compliance.yml@main
    name: reviewdog
    on: [pull_request]
    workflow_dispatch:

  vale:
    name: runner / vale
    runs-on: ubuntu-latest
    steps:
      - name: checkout
        uses: actions/checkout@v3
      - name: linting
        uses: errata-ai/vale-action@reviewdog
        files: "* .md"
       
```
What's weird is that I got this result when I copied the code in a YML validator

yaml-checker-results

But got these results in my terminal 

from my terminal
Hopefully, things will be better tomorrow 

### Update 10/30/23
I reread the documentation for the GitHub Action and apparently, it is recommended to follow this repo style:

```
.github
├── styles
│   └── vocab.txt
└── workflows
    └── main.yml
.vale.ini
...
```
Now the hard part is figuring out what to put in the `vocab.txt`. 🤔 

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

## [optional] Are there any post-deployment tasks we need to perform?

Once the PR is merged, make sure to click on Read and write permissions in action settings.

### CODE REVIEW

**Comment by nickytonline** [comment]
> What's weird is that I got this result when I copied the code in a YML validator
> 
> yaml-checker-results
> 
> But got these results in my terminal 
> 
> from my terminal
Hopefully, things will be better tomorrow 

It's valid YAML, but it doesn't seem to be correct for what a GitHub action expects. I'd double check the Vale linter docs and also compare it to some existing actions in the app repo in terms of GitHub action syntax.

File: .github/workflows/compliance.yml [hunk 1]
```diff
@@ -13,3 +13,14 @@ permissions:
...
+on: [pull_request]
+
+jobs:
+  vale:
+    name: runner / vale
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: errata-ai/vale-action@reviewdog
+         files: "* .md"
```
Comment (dlemstra) [inline .github/workflows/compliance.yml:26]: This should be:

```yaml
- uses: errata-ai/vale-action@reviewdog
  with:
    files: "* .md"
```

And it looks like the value for files should be a path: https://github.com/errata-ai/vale-action#files-default-all

File: .github/workflows/compliance.yml [hunk 2]
```diff
@@ -13,3 +13,14 @@ permissions:
 jobs:
   compliance:
     uses: open-sauced/hot/.github/workflows/compliance.yml@main
+name: reviewdog
```
Comment (nickmccurdy) [inline .github/workflows/compliance.yml:16]: This already has a name on line 1, did you mean to edit it?

Comment (CBID2) [reply .github/workflows/compliance.yml:16]: No @nickmccurdy.

Comment (nickmccurdy) [reply .github/workflows/compliance.yml:16]:
```suggestion
```

**Comment by BekahHW** [comment]
@CBID2 how's this going?

**Comment by BekahHW** [comment]
@CBID2 when you get a chance, share what you've tried and where you've looked in their docs and maybe we can point you in the right direction. Also, let us know how you're testing it.

**Comment by BekahHW** [comment]
Hey @CBID2 are you still working on this issue?

**Comment by CBID2** [comment]
> Hey @CBID2 are you still working on this issue?

Yeah I am @BekahHW. I just learned about this in Vale's documentation: https://vale.sh/generator/
Which one do you think would be best suited for this repo?

**Comment by BekahHW** [comment]
> > Hey @CBID2 are you still working on this issue?
> 
> Yeah I am @BekahHW. I just learned about this in Vale's documentation: [vale.sh/generator](https://vale.sh/generator/) Which one do you think would be best suited for this repo?

Can you explain what that is and why we need it? I need some more context before I look at it.

**Comment by CBID2** [comment]
> > > Hey @CBID2 are you still working on this issue?
> 
> > 
> 
> > Yeah I am @BekahHW. I just learned about this in Vale's documentation: [vale.sh/generator](https://vale.sh/generator/) Which one do you think would be best suited for this repo?
> 
> 
> 
> Can you explain what that is and why we need it? I need some more context before I look at it.

It's a generator that creates config files based on the style guides people choose (e.g., Google and Microsoft) and the static generators that is used (e.g., Hugo). Does that make more sense now @BekahHW?

**Comment by nickytonline** [comment]
I'm going to go ahead and close this as it's been open since Oct 2023. Feel free to reopen if you plan to continue working on this.

**Comment by CBID2** [comment]
> I'm going to go ahead and close this as it's been open since Oct 2023. Feel free to reopen if you plan to continue working on this.

Hey @nickytonline. I think I found a possible solution.  Can you make a PR on my test repo again: https://github.com/CBID2/github-actions-test-for-vale

**Comment by BekahHW** [comment]
Hey @CBID2, If you can't get this resolved in the next week, let's let someone else try taking on the issue, since you've had it since October.

**Comment by CBID2** [comment]
> Hey @CBID2, If you can't get this resolved in the next week, let's let someone else try taking on the issue, since you've had it since October.

Hi @BekahHW. I made some new updates to my test repo, and was hoping that Nick would do another PR to it but I never got a response. Can you try it?

**Comment by BekahHW** [comment]
I saw that Ryan submitted a PR that failed. That's why I didn't.

**Comment by CBID2** [comment]
> I saw that Ryan submitted a PR that failed. That's why I didn't. 

Thanks for pointing it out @BekahHW. Just checked it out

**Comment by CBID2** [comment]
Hey @BekahHW. To update you, I worked on my test branch with someone at Virtual Coffee and here's a solution we came up with: https://github.com/CBID2/github-actions-test-for-vale/pull/10/checks
As you can see, the action does point out the file's errors, but the checkmark in the runner/vale test remains green. Do you recommend that I put this approach in this PR and then create an issue for someone to help the action show red for the vale test?

**Comment by nickytonline** [comment]
There's no point in adding it if it's only partially working.

I don't see where it's reporting the file errors in what you linked. I see changes to the GitHub action file in that PR, but that's it.

It should look like this like they have in their docs.



Do you have a test PR in your repo showing where it displays lint errors in some text?

**Comment by CBID2** [comment]
> There's no point in adding it if it's only partially working.
> 
> I don't see where it's reporting the file errors in what you linked. I see changes to the GitHub action file in that PR, but that's it.
> 
> It should look like this like they have in their docs.
> 
> 
> 
> Do you have a test PR in your repo showing where it displays lint errors in some text?

Here's what I am referring to @nickytonline 
errors
To answer your question, no

**Comment by CBID2** [comment]
Hi @nickytonline! :) I have good news! :) I managed to get Vale to work via this PR: https://github.com/CBID2/github-actions-test-for-vale/pull/13#issuecomment-1973649001 
What I have to do now is make it show up in the text as shown in the picture you posted.

**Comment by CBID2** [comment]
Hey @nickytonline. Somehow in my test repo, the errors have appeared in the log but the annotations have appeared on the present file and not the one I’m pushing. I raised [an issue about this to Vale’s repo](https://github.com/errata-ai/vale-action/issues/120), so hopefully, I'll get a response

**Pull request closed without merging**